enterprise/server/cmd/ci_runner/main.go

```diff
@@ -803,12 +885,45 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace) error {
 		return ws.setupError
 	}
 
+	if ul := ar.reporter.uploader; ul != nil {
+		allUploadsChan := make(chan []*bes_artifact_uploader.Result)
+		// Buffer upload results
+		go func() {
+			defer close(allUploadsChan)
+			var uploads []*bes_artifact_uploader.Result
+			for u := range ar.reporter.uploader.Results() {
+				uploads = append(uploads, u)
+			}
+			allUploadsChan <- uploads
+		}()
```

Comment:
Is this channel necessary?

Reply:
Got rid of it in favor of having the Uploader be responsible for buffering all the upload result metadata.